src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -256,17 +258,10 @@ func (r *ClusterReconciler) Reconcile(
 		secrets = append(secrets, schemaRegistrySu.Key())
 	}
 
-	err = r.setInitialSuperUserPassword(ctx, adminAPI, secrets)
-
-	var e *resources.RequeueAfterError
-	if errors.As(err, &e) {
-		log.Info(e.Error())
-		return ctrl.Result{RequeueAfter: e.RequeueAfter}, nil
-	}
-
-	if err != nil {
-		log.Error(err, "Failed to set up initial super user password")
-		return ctrl.Result{}, err
+	if errSetInit := r.setInitialSuperUserPassword(ctx, adminAPI, secrets); errSetInit != nil {
+		// we capture all errors here, do not return the error, just requeue
+		log.Info(errSetInit.Error())
+		return ctrl.Result{RequeueAfter: resources.RequeueDuration}, nil
```

Comment:
Returning the error implements the exponential backoff. Returning a requeue resets that timer. Wouldn't we want the exponential backoff here?

Reply:
I agree, but that's not what the original code was doing in general, here i shortcut the original intent and obviously a concern bubbles up :)